File: Lab4/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.widget.Button;
import android.widget.GridView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_layout);
        ArrayList<Button> buttons = new ArrayList<>();
        int maxButtons = 3;
        for (int i = 1; i <= maxButtons; i++) {
            Button newButton = new Button(this);
            buttons.add(newButton);
        }
        ButtonAdapter buttonAdapter = new ButtonAdapter(this, R.layout.menurecylerrow, buttons);
        GridView gridView = findViewById(R.id.exList);
        gridView.setAdapter(buttonAdapter);
    }
}

File: Lab6/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.GridView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val buttons = ArrayList<Button>()
        val maxButtons = 3
        for (i in 1..maxButtons) {
            val newButton = Button(this)
            buttons.add(newButton)
        }
        val buttonAdapter = ButtonAdapter(this, R.layout.menurecylerrow, buttons)
        val gridView = findViewById<GridView>(R.id.exList)
        gridView.adapter = buttonAdapter
    }
}